src/accounts/channel/set-channel-account-command.js

```diff
@@ -0,0 +1,87 @@
+const Command = require('../../core/commands/command');
+const commandRegexWithAlias = /set channel account (.*)/;
+const commandRegex = /set channel account/;
+const LoggerFactory = require('../../core/logging/logger-factory');
+const logger = LoggerFactory.getLogger(__filename);
+const { getEventMetadata } = require('../../core/logging/logging-metadata');
+
+class SetChannelAccountCommand extends Command {
+  constructor(channelHandler) {
+    super();
+    this.channelHandler = channelHandler;
+  }
+  configure(controller) {
+    controller.hears(
+      [commandRegexWithAlias],
+      'direct_message,direct_mention',
+      (bot, message) => {
+        let alias = message.match[1];
+        this.setChannel(message, bot, alias);
+      }
+    );
+
+    controller.hears(
+      [commandRegex],
+      'direct_message,direct_mention',
+      (bot, message) => {
+        this.ask(bot, message.user, message.team, message);
+      }
```

Comment:
if message contain the user and the team
why not to pass only the message to the ask 
in fact u can even do :
controller.hears(
      [commandRegex],
      'direct_message,direct_mention',
      this.ask)

and then ask function should look like:

ask(bot , message) {
const {user, team} = message;
.....
}

Reply:
resolved